Upsun Quick start guide (#3554)
platformsh/platformsh-docs: sites/friday/src/get-started/here/configure/nodejs.md

---
title: JavaScript/Node.js
weight: 10
# description: All you need to know about creating a new project with {{% vendor/name %}}
---

When dealing with Javascript/Node.js stacks, the information below may help customize your configuration.
These sections provide Javascript/Node.js-specific configuration details, but also be sure to consult the common Upsun documentation as well:

- [Configuring applications](/create-apps)
- [Setting up managed services](/add-services)
- [Handling requests](/define-routes)

## Build flavors

> By default, Upsun will make assumptions about how you want to build your application. 

Reviewer: ```suggestion
By default, Upsun makes assumptions about how you want to build your application. 
```

Namely, that you are managing your dependencies with npm, and that the very first thing you'd like to run is a particular and common production flavor of `npm install`.

This is called a build `flavor`, but it's assumption may prove inappropriate for your application and cause your builds to fail (such as if you'd like to use yarn or bun instead of npm).
[Consult the documentation](/languages/nodejs#dependencies) to learn more about disabling this feature.

## Available package managers

Certain package managers come pre-installed on all Upsun `nodejs` container types:

* *npm*; example: ``npm install platformsh-config``
* *npx*; example: ``npx create-strapi-app <APP_NAME>``
* *bun*; example: ``bun install platformsh-config``

> Yarn is an exception, as it [requires that you install it explicitly](/languages/nodejs#use-yarn-as-a-package-manager). 

Reviewer 2: Yarn **is** bundled with node via corepack but isn't _accessible_ expect through the use of corepack. Users dont have to install it but they do have to explicitly declare its use in the package.json file.

      nodejs:
        sharp: "*"
#services:
#  db:
#    type: postgresql:14
routes:
  "https://{default}/":
    type: upstream
    upstream: "app:http"
  # A basic redirect definition
  # More information: {{% vendor/url_doc %}}/define-routes.html#basic-redirect-definition
  "https://www.{default}/":
    type: redirect
    to: "https://{default}/"
```

<--->
+++
title=Strapi
+++

```yaml {location=".{{% vendor/cli %}}/config.yaml"}
applications:
  app:
    source:
      root: "/"
    type: "nodejs:20"
    relationships:
      postgresql: "postgresql:postgresql"
    mounts:
      ...
    web:
      commands:
        start: "NODE_ENV=production yarn start"
    build:
      flavor: none
    dependencies:
      nodejs:
        yarn: "^1.22.0"
    hooks:
      build: |
        set -eux
        yarn
        yarn build
services:
  postgresql:
    type: postgresql:15

routes:
  "https://{default}/":
    type: upstream
    upstream: "app:http"
  # A basic redirect definition
  # More information: {{% vendor/url_doc %}}/define-routes.html#basic-redirect-definition
  "https://www.{default}/":
    type: redirect
    to: "https://{default}/"
```

{{< /codetabs >}}

## Frameworks

> The Upsun documentation includes a wide array of community resources that will help with framework-specific configuration:

Reviewer: ```suggestion
The Upsun documentation includes a wide array of community resources to help with framework-specific configuration:
```
